File: ios/ContractRunner/Sources/ContractRunner/ProfileSorter.swift
import Foundation

public struct ProfileSortInput: Equatable {
    public let id: String
    public let name: String
    public let isKids: Bool
    public let lastUsedMs: Int64?

    public init(id: String, name: String, isKids: Bool, lastUsedMs: Int64?) {
        self.id = id
        self.name = name
        self.isKids = isKids
        self.lastUsedMs = lastUsedMs
    }
}

public func sortProfiles(_ profiles: [ProfileSortInput]) -> [String] {
    let sorted = profiles.sorted { lhs, rhs in
        if lhs.isKids != rhs.isKids {
            return lhs.isKids && !rhs.isKids
        }
        let lhsUsed = lhs.lastUsedMs ?? Int64.min
        let rhsUsed = rhs.lastUsedMs ?? Int64.min
        if lhsUsed != rhsUsed {
            return lhsUsed > rhsUsed
        }
        return lhs.id < rhs.id
    }
    return sorted.map { $0.id }
}

File: android/core-domain/src/main/kotlin/com/crispy/tv/domain/account/ProfileSorter.kt
package com.crispy.tv.domain.account

data class ProfileSortInput(
    val id: String,
    val name: String,
    val isKids: Boolean,
    val lastUsedMs: Long?
)

fun sortProfiles(profiles: List<ProfileSortInput>): List<String> {
    return profiles
        .sortedWith(
            compareByDescending<ProfileSortInput> { it.isKids }
                .thenByDescending { it.lastUsedMs ?: Long.MIN_VALUE }
                .thenBy { it.id }
        )
        .map { it.id }
}
